src/idl_gen_rust.cpp

```diff
@@ -2529,6 +2691,17 @@ class RustGenerator : public BaseGenerator {
       ForAllStructFields(struct_def, [&](const FieldDef &field) {
         if (IsStruct(field.value.type)) {
           code_ += "      &self.{{FIELD_NAME}}.pack(),";
+        } else if (IsArray(field.value.type)) {
+          if (GetFullType(field.value.type) == ftArrayOfStruct) {
+            code_.SetValue("ARRAY_ITEM",
+                           GetTypeGet(field.value.type.VectorType()));
+            code_ +=
+                "      &self.{{FIELD_NAME}}.iter()"
+                ".map(|x| x.pack()).collect::<Vec<{{ARRAY_ITEM}}>>()"
```

Comment:
probably should use a for loop in this place to avoid the allocation from Vec

Reply:
This is done similarly today for vectors, first collect into a native vector and then create flatbuffers vector (line 2115).
If we want to make it more optimal for array I see a few options:
1. Pretty much copy all the `into` implementation when we transform an array of native structs, which will create a lot of bloat in my opinion.
2. Use a crate array-init. Which is much easier and cleaner but will require adding another crate to flatbuffer dependencies and I don't know what is the policy for that.

Reply:
I think it's different because we can't know the size of a vector statically, but we can for arrays.
I don't think we should take the dependency but to avoid bloat we could put some function in the flatbuffers crate for packing and unpacking arrays

Reply:
Hmm, I see what you mean.
Ok, I will port the relevant implementation from array-init